Makefile

```diff
@@ -171,7 +182,7 @@ LDFLAGS:=-ldflags "\
 .PHONY: build
 build: bin/calico-typha
 bin/calico-typha: bin/calico-typha-$(ARCH)
-bin/calico-typha-$(ARCH): $(TYPHA_GO_FILES) vendor/.up-to-date
+bin/calico-typha-$(ARCH): $(TYPHA_GO_FILES) register vendor/.up-to-date
```

Comment:
You shouldn't depend on a phony target, that'll cause the target to always be rebuilt.  I'd probably do this:
```
bin/calico-typha-$(ARCH): $(TYPHA_GO_FILES) vendor/.up-to-date
        $(MAKE) register
        ...
```
It has the downside that register can be run more than once if more than one target calls it but I'm guessing it only takes a fraction of a second anyway.

Reply:
I don't get it. What is the difference between a phony target that is a dependency of `bin/calico-typha-$(ARCH)`, and calling `$(MAKE) register` in the same target? Either way, it always will be run?

Reply:
Sorry, I meant: if you depend on a phony, `bin/calico-typha-$(ARCH)` will always be considered stale and `bin/calico-typha-$(ARCH)` will always be rebuilt.  Right now, if you do 
```
make bin/calico-typha-amd64
```
then it'll only be rebuilt if one of the `.go` files or `vendor/.up-to-date` has changed.

If you re-run the phony using `$(MAKE) register` then make will correctly detect the staleness of `bin/calico-typha-amd64` and `$(MAKE) register` will only get run if `bin/calico-typha-amd64` is stale (Which is almost[1] what we want).  

[1] Unfortunately, `$(MAKE) register` might be run more than once if more than one target uses it but the impact of that is probably very small.

Reply:
Done, and pushed out.